File: src/main/cpp/commands/cGroup_Elevator.cpp
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

#include "commands/cGroup_Elevator.h"
#include <frc2/command/Commands.h>
#include <frc/smartdashboard/Smartdashboard.h>
#include <frc2/command/WaitCommand.h>
// NOTE:  Consider using this command inline, rather than writing a subclass.
// For more information, see:
// https://docs.wpilib.org/en/stable/docs/software/commandbased/convenience-features.html

// Teleop Commands
  frc2::CommandPtr ElevatorMovements::ScoreHighCone(subsystem_Elevator* Elevator, subsystem_Wrist* Wrist, subsystem_Intake* Intake, subsystem_LED* LED){
    return frc2::cmd::Sequence(command_MoveWrist(Wrist, [=]{return 0;}, [=]{return true;}, [=]{return WristConstants::WristThreshold;}).ToPtr(),
                                command_MoveElevator(Elevator, [=]{return ElevatorConstants::kElevatorScoreHighCone;}, [=]{return false;}, [=]{return ElevatorConstants::ElevatorThreshold;}).ToPtr(),
                                command_MoveWrist(Wrist, [=]{return WristConstants::HighConePos;}, [=]{return true;}, [=]{return WristConstants::WristThreshold;}).ToPtr(),
                                command_Outtake(Intake, LED).ToPtr());
  }
  frc2::CommandPtr ElevatorMovements::ScoreMidCone(subsystem_Elevator* Elevator, subsystem_Wrist* Wrist, subsystem_Intake* Intake, subsystem_LED* LED){
    return frc2::cmd::Sequence(command_MoveWrist(Wrist, [=]{return 0;}, [=]{return true;}, [=]{return WristConstants::WristThreshold;}).ToPtr(),
                                command_MoveElevator(Elevator, [=]{return ElevatorConstants::kElevatorScoreMidCone;}, [=]{return false;}, [=]{return ElevatorConstants::ElevatorThreshold;}).ToPtr(),
                                command_MoveWrist(Wrist, [=]{return WristConstants::MidConePos;}, [=]{return true;}, [=]{return WristConstants::WristThreshold;}).ToPtr(),
                                command_Outtake(Intake, LED).ToPtr());
  }
  frc2::CommandPtr ElevatorMovements::ScoreHighCube(subsystem_Elevator* Elevator, subsystem_Wrist* Wrist, subsystem_Intake* Intake, subsystem_LED* LED){
    return frc2::cmd::Sequence(command_MoveWrist(Wrist, [=]{return 0;}, [=]{return true;}, [=]{return WristConstants::WristThreshold;}).ToPtr(),
                                command_MoveElevator(Elevator, [=]{return ElevatorConstants::kElevatorScoreHighCube;}, [=]{return false;}, [=]{return ElevatorConstants::ElevatorThreshold;}).ToPtr(),
                                command_MoveWrist(Wrist, [=]{return WristConstants::HighCubePos;}, [=]{return true;}, [=]{return WristConstants::WristThreshold;}).ToPtr(),
                                command_Outtake(Intake, LED).ToPtr());
  }
  frc2::CommandPtr ElevatorMovements::ScoreMidCube(subsystem_Elevator* Elevator, subsystem_Wrist* Wrist, subsystem_Intake* Intake, subsystem_LED* LED){
    return frc2::cmd::Sequence(command_MoveWrist(Wrist, [=]{return 0;}, [=]{return true;}, [=]{return WristConstants::WristThreshold;}).ToPtr(),
                                command_MoveElevator(Elevator, [=]{return ElevatorConstants::kElevatorScoreMidCube;}, [=]{return false;}, [=]{return ElevatorConstants::ElevatorThreshold;}).ToPtr(),
                                command_MoveWrist(Wrist, [=]{return WristConstants::MidCubePos;}, [=]{return true;}, [=]{return WristConstants::WristThreshold;}).ToPtr(),
                                command_Outtake(Intake, LED).ToPtr());
  }
  frc2::CommandPtr ElevatorMovements::ScoreHybrid(subsystem_Elevator* Elevator, subsystem_Wrist* Wrist, subsystem_Intake* Intake, subsystem_LED* LED){
    return frc2::cmd::Sequence(command_MoveWrist(Wrist, [=]{return 0;}, [=]{return true;}, [=]{return WristConstants::WristThreshold;}).ToPtr(),
                                command_MoveElevator(Elevator, [=]{return ElevatorConstants::kElevatorScoreHybrid;}, [=]{return false;}, [=]{return ElevatorConstants::ElevatorThreshold;}).ToPtr(),
                                command_MoveWrist(Wrist, [=]{return WristConstants::GroundIntake;}, [=]{return true;}, [=]{return WristConstants::WristThreshold;}).ToPtr(),
                                command_Outtake(Intake, LED).ToPtr());
  }
  frc2::CommandPtr ElevatorMovements::ToStow(subsystem_Elevator* Elevator, subsystem_Wrist* Wrist, subsystem_Intake* Intake, subsystem_LED* LED){
    return frc2::cmd::Sequence(command_MoveWrist(Wrist, [=]{return 0;}, [=]{return true;}, [=]{return WristConstants::WristThreshold;}).ToPtr(),
                                command_MoveElevator(Elevator, [=]{return ElevatorConstants::kElevatorStow;}, [=]{return false;}, [=]{return ElevatorConstants::ElevatorThreshold;}).ToPtr(),
                                Wrist->ZeroWrist());
  }

  frc2::CommandPtr ElevatorMovements::GroundIntake(subsystem_Elevator* Elevator, subsystem_Wrist* Wrist, subsystem_Intake* Intake, subsystem_LED* LED){
    return frc2::cmd::Sequence(command_MoveWrist(Wrist, [=]{return 0;}, [=]{return true;}, [=]{return WristConstants::WristThreshold;}).ToPtr(),
                                command_MoveElevator(Elevator, [=]{return ElevatorConstants::kElevatorScoreHybrid;}, [=]{return false;}, [=]{return ElevatorConstants::ElevatorThreshold;}).ToPtr(),
                                command_MoveWrist(Wrist, [=]{return WristConstants::GroundIntake;}, [=]{return true;}, [=]{return WristConstants::WristThreshold;}).ToPtr(),
                                command_Intake(Intake, LED).ToPtr());
  }
  frc2::CommandPtr ElevatorMovements::DoubleSubstationIntake(subsystem_Elevator* Elevator, subsystem_Wrist* Wrist, subsystem_Intake* Intake, subsystem_LED* LED){
    return frc2::cmd::Sequence(command_MoveWrist(Wrist, [=]{return 0;}, [=]{return true;}, [=]{return WristConstants::WristThreshold;}).ToPtr(),
                                command_MoveElevator(Elevator, [=]{return ElevatorConstants::kElevatorIntakeDoubleSubstation;}, [=]{return false;}, [=]{return ElevatorConstants::ElevatorThreshold;}).ToPtr(),
                                command_MoveWrist(Wrist, [=]{return WristConstants::SubstationPos;}, [=]{return true;}, [=]{return WristConstants::WristThreshold;}).ToPtr(),
                                command_Intake(Intake, LED).ToPtr());
  }

  // Auton Commands
  frc2::CommandPtr ElevatorMovements::ScoreHighConeStow(subsystem_Elevator* Elevator, subsystem_Wrist* Wrist, subsystem_Intake* Intake, subsystem_LED* LED){
    return frc2::cmd::Sequence(ElevatorMovements::ScoreHighCone(Elevator, Wrist, Intake, LED), ElevatorMovements::ToStow(Elevator, Wrist, Intake, LED));
  }
  frc2::CommandPtr ElevatorMovements::ScoreMidConeStow(subsystem_Elevator* Elevator, subsystem_Wrist* Wrist, subsystem_Intake* Intake, subsystem_LED* LED){
    return frc2::cmd::Sequence(ElevatorMovements::ScoreMidCone(Elevator, Wrist, Intake, LED), ElevatorMovements::ToStow(Elevator, Wrist, Intake, LED));
  }
  frc2::CommandPtr ElevatorMovements::ScoreHighCubeStow(subsystem_Elevator* Elevator, subsystem_Wrist* Wrist, subsystem_Intake* Intake, subsystem_LED* LED){
    return frc2::cmd::Sequence(ElevatorMovements::ScoreHighCube(Elevator, Wrist, Intake, LED), ElevatorMovements::ToStow(Elevator, Wrist, Intake, LED));
  }
  frc2::CommandPtr ElevatorMovements::ScoreMidCubeStow(subsystem_Elevator* Elevator, subsystem_Wrist* Wrist, subsystem_Intake* Intake, subsystem_LED* LED){
    return frc2::cmd::Sequence(ElevatorMovements::ScoreMidCube(Elevator, Wrist, Intake, LED), ElevatorMovements::ToStow(Elevator, Wrist, Intake, LED));
  }

File: src/main/java/frc/robot/commands/cGroup_Elevator.java
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

package frc.robot.commands;

import frc.robot.subsystems.*;
import edu.wpi.first.wpilibj2.command.Commands;
import edu.wpi.first.wpilibj2.command.Command;
import frc.robot.Constants.*;

public class cGroup_Elevator{
  public static Command scoreHighCone(subsystem_Elevator elevator, subsystem_Wrist wrist, subsystem_Intake intake, subsystem_LEDs led){
    return Commands.sequence(
      new command_MoveWrist(wrist, () -> {return 0;}, () -> {return true;}, () -> {return WristConstants.wristThreshold;}),
      new command_MoveElevator(elevator, () -> {return ElevatorConstants.kElevatorScoreHighCone;}, () -> {return false;}, () -> {return ElevatorConstants.elevatorThreshold;}),
      new command_MoveWrist(wrist, () -> {return WristConstants.highConePos;}, () -> {return true;}, () -> {return WristConstants.wristThreshold;}),
      new command_Outtake(intake, led)
    );
  }

  public static Command scoreMidCone(subsystem_Elevator elevator, subsystem_Wrist wrist, subsystem_Intake intake, subsystem_LEDs led){
    return Commands.sequence(
      new command_MoveWrist(wrist, () -> {return 0;}, () -> {return true;}, () -> {return WristConstants.wristThreshold;}),
      new command_MoveElevator(elevator, () -> {return ElevatorConstants.kElevatorScoreMidCone;}, () -> {return false;}, () -> {return ElevatorConstants.elevatorThreshold;}),
      new command_MoveWrist(wrist, () -> {return WristConstants.midConePos;}, () -> {return true;}, () -> {return WristConstants.wristThreshold;}),
      new command_Outtake(intake, led)
    );
  }

  public static Command scoreHighCube(subsystem_Elevator elevator, subsystem_Wrist wrist, subsystem_Intake intake, subsystem_LEDs led){
    return Commands.sequence(
      // new command_MoveWrist(wrist, () -> {return 0;}, () -> {return true;}, () -> {return WristConstants.wristThreshold;}),
      wrist.zeroWrist(),
      new command_MoveElevator(elevator, () -> {return ElevatorConstants.kElevatorScoreHighCube;}, () -> {return false;}, () -> {return ElevatorConstants.elevatorThreshold;}),
      new command_MoveWrist(wrist, () -> {return WristConstants.highCubePos;}, () -> {return true;}, () -> {return WristConstants.wristThreshold;}),
      new command_Outtake(intake, led)
    );
  }

  public static Command scoreMidCube(subsystem_Elevator elevator, subsystem_Wrist wrist, subsystem_Intake intake, subsystem_LEDs led){
    return Commands.sequence(
      wrist.zeroWrist(),
      new command_MoveElevator(elevator, () -> {return ElevatorConstants.kElevatorScoreMidCube;}, () -> {return false;}, () -> {return ElevatorConstants.elevatorThreshold;}),
      new command_MoveWrist(wrist, () -> {return WristConstants.midCubePos;}, () -> {return true;}, () -> {return WristConstants.wristThreshold;}),
      new command_Outtake(intake, led)
    );
  }

  public static Command scoreHybrid(subsystem_Elevator elevator, subsystem_Wrist wrist, subsystem_Intake intake, subsystem_LEDs led){
    return Commands.sequence(
      new command_MoveWrist(wrist, () -> {return 0;}, () -> {return true;}, () -> {return WristConstants.wristThreshold;}),
      new command_MoveElevator(elevator, () -> {return ElevatorConstants.kElevatorScoreHybrid;}, () -> {return false;}, () -> {return ElevatorConstants.elevatorThreshold;}),
      new command_MoveWrist(wrist, () -> {return WristConstants.groundIntake;}, () -> {return true;}, () -> {return WristConstants.wristThreshold;}),
      new command_Outtake(intake, led)
    );
  }

  public static Command toStow(subsystem_Elevator elevator, subsystem_Wrist wrist){
    return Commands.sequence(
      new command_MoveWrist(wrist, () -> {return 0;}, () -> {return true;}, () -> {return WristConstants.wristThreshold;}),
      new command_MoveElevator(elevator, () -> {return ElevatorConstants.kElevatorStow;}, () -> {return false;}, () -> {return ElevatorConstants.elevatorThreshold;}),
      wrist.zeroWrist()
      );
  }

  public static Command groundIntake(subsystem_Elevator elevator, subsystem_Wrist wrist, subsystem_Intake intake, subsystem_LEDs led){
    return Commands.sequence(new command_MoveWrist(wrist, () -> 0, () -> true, () -> WristConstants.wristThreshold),
                            new command_MoveElevator(elevator, () -> ElevatorConstants.kElevatorScoreHybrid, () -> false, () -> ElevatorConstants.elevatorThreshold),
                            new command_MoveWrist(wrist, () -> WristConstants.groundIntake, () -> true, () -> WristConstants.wristThreshold),
                            new command_Intake(intake, led));
  }

  public static Command doubleSubstationIntake(subsystem_Elevator elevator, subsystem_Wrist wrist, subsystem_Intake intake, subsystem_LEDs led){
    return Commands.sequence(new command_MoveWrist(wrist, () -> 0, () -> true, () -> WristConstants.wristThreshold),
                            new command_MoveElevator(elevator, () -> ElevatorConstants.kElevatorIntakeDoubleSubstation, () -> false, () -> ElevatorConstants.elevatorThreshold),
                            new command_MoveWrist(wrist, () -> WristConstants.substationPos, () -> true, () -> WristConstants.wristThreshold),
                            new command_Intake(intake, led));
  }

  // Auton Commands
  public static Command scoreHighConeStow(subsystem_Elevator elevtor, subsystem_Wrist wrist, subsystem_Intake intake, subsystem_LEDs led){
    return Commands.sequence(scoreHighCone(elevtor, wrist, intake, led), toStow(elevtor, wrist));
  }
  public static Command scoreMidConeStow(subsystem_Elevator elevtor, subsystem_Wrist wrist, subsystem_Intake intake, subsystem_LEDs led){
    return Commands.sequence(scoreMidCone(elevtor, wrist, intake, led), toStow(elevtor, wrist));
  }
  public static Command scoreHighCubeStow(subsystem_Elevator elevtor, subsystem_Wrist wrist, subsystem_Intake intake, subsystem_LEDs led){
    return Commands.sequence(scoreHighCube(elevtor, wrist, intake, led), toStow(elevtor, wrist));
  }
  public static Command scoreMidCubeStow(subsystem_Elevator elevtor, subsystem_Wrist wrist, subsystem_Intake intake, subsystem_LEDs led){
    return Commands.sequence(scoreMidCube(elevtor, wrist, intake, led), toStow(elevtor, wrist));
  }
}
// NOTE:  Consider using this command inline, rather than writing a subclass.  For more
// information, see:
// https://docs.wpilib.org/en/stable/docs/software/commandbased/convenience-features.html